src/ln/channel.rs

```diff
@@ -2472,6 +2486,22 @@ impl Channel {
 			return Err(ChannelError::Close("Peer sent a garbage channel_reestablish"));
 		}
 
+		if msg.next_remote_commitment_number > 0 {
+			match msg.data_loss_protect {
+				OptionalField::Present(ref data_loss) => {
+					if chan_utils::build_commitment_secret(self.local_keys.commitment_seed, INITIAL_COMMITMENT_NUMBER - msg.next_remote_commitment_number + 1) != data_loss.your_last_per_commitment_secret {
+						return Err(ChannelError::Close("Peer sent a garbage channel_reestablish with secret key not matching the commitment height provided"));
+					}
+					if INITIAL_COMMITMENT_NUMBER - msg.next_remote_commitment_number < self.cur_local_commitment_transaction_number {
+						self.channel_monitor.provide_latest_remote_commitment_tx_info(&Transaction { version: 2, lock_time: 0, input: Vec::new(), output: Vec::new(), }, Vec::new(), INITIAL_COMMITMENT_NUMBER - msg.next_remote_commitment_number, data_loss.my_current_per_commitment_point);
+						return Err(ChannelError::Stale { msg: "FALLEN-BEHIND: we need to stale this channel and wait than our up-to-date non-malicious peer broadcast its own commitment transaction on chain", update: Some(self.channel_monitor.clone())
```

Comment:
"stale" is not a verb. Maybe "We have fallen behind - we need to close the channel and give our peer some time to (hopefully) broadcast the latest commitment transaction is has, but should give up at some point and broadcast our own."
Also, I don't think we need to capitalize FALLEN-BEHIND, I don't think there's anything else like that in the codebase.

Reply:
Corrected, but there is not that much place in our codebase where we learn something is completely wrong and user should be careful.